### ISSUE TITLE
Request: [SQL Server] Provide better exception message when table mapping is wrong

### ISSUE THREAD

**Issue by GeeWee** [mikependon/RepoDB#443]
If you have an entity you try to insert into an SQL server 2019 database, and the table is wrong, you'll end up with an error message that doesn't really tell you what's wrong.

E.g. take the following entity
```csharp
public class AlarmCodeEntity
{
    public int Id { get; set; }
    // some other fields
}
```
But the following table
```sql
create table dbo.AlarmCode
(
Id  int identity
constraint PK_AlarmCode
   primary key,
-- Some other columns
)
```
(notice the name mismatch - the entity is called `AlarmCodeEntity`, and the table is called `AlarmCode`

You get the following stack-trace when trying to insert an entity.
```
Error Message:
   Microsoft.Data.SqlClient.SqlException : Must declare the scalar variable "@Id".
Must declare the scalar variable "@Id".
  Stack Trace:
     at Microsoft.Data.SqlClient.SqlConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at Microsoft.Data.SqlClient.SqlInternalConnection.OnError(SqlException exception, Boolean breakConnection, Action`1 wrapCloseInAction)
   at Microsoft.Data.SqlClient.TdsParser.ThrowExceptionAndWarning(TdsParserStateObject stateObj, Boolean callerHasConnectionLock, Boolean asyncClose)
   at Microsoft.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, SqlCommand cmdHandler, SqlDataReader dataStream, BulkCopySimpleResultSet bulkCopyHandler, TdsParserStateObject stateObj, Boolean& dataReady)
   at Microsoft.Data.SqlClient.SqlDataReader.TryConsumeMetaData()
   at Microsoft.Data.SqlClient.SqlDataReader.get_MetaData()
   at Microsoft.Data.SqlClient.SqlCommand.FinishExecuteReader(SqlDataReader ds, RunBehavior runBehavior, String resetOptionsString, Boolean isInternal, Boolean forDescribeParameterEncryption, Boolean shouldCacheForAlwaysEncrypted)
   at Microsoft.Data.SqlClient.SqlCommand.RunExecuteReaderTds(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, Boolean isAsync, Int32 timeout, Task& task, Boolean asyncWrite, Boolean inRetry, SqlDataReader ds, Boolean describeParameterEncryptionRequest)
   at Microsoft.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, TaskCompletionSource`1 completion, Int32 timeout, Task& task, Boolean& usedCache, Boolean asyncWrite, Boolean inRetry, String method)
   at Microsoft.Data.SqlClient.SqlCommand.RunExecuteReader(CommandBehavior cmdBehavior, RunBehavior runBehavior, Boolean returnStream, String method)
   at Microsoft.Data.SqlClient.SqlCommand.ExecuteScalar()
   at RepoDb.DbConnectionExtension.InsertAsyncInternalBase[TEntity,TResult](IDbConnection connection, String tableName, TEntity entity, IEnumerable`1 fields, String hints, Nullable`1 commandTimeout, IDbTransaction transaction, ITrace trace, IStatementBuilder statementBuilder, Boolean skipIdentityCheck)
```
Now the correct thing to do is to use the `[Map]` attribute, and if the following is added to the entity it works fine:
`[Map("[dbo].AlarmCode")]`
However this is not very clear from the error message. I was wondering if there was some way we could improve that.

**Comment by mikependon** [comment]
Of course, I also saw this. :) For now, as you mentioned, you can either use any of the following below to fix the issue.

- `Table()`
- `Map()`
- `FluentMapper.Entity<AlarmCodeEntity>().Table("AlarmCode")`.

**Comment by mikependon** [comment]
This is now available at RepoDb >= v1.11.2.

**Comment by GeeWee** [comment]
I can confirm that the error messages are now much nicer!
